erizo_controller/erizoJS/models/Publisher.js

```diff
@@ -129,24 +125,169 @@ class Source extends Node {
     return this.externalOutputs[url];
   }
 
-  muteStream(muteVideo, muteAudio) {
-    this.muteVideo = muteVideo;
-    this.muteAudio = muteAudio;
-    for (var subId in this.subscribers) {
-      var sub = this.getSubscriber(subId);
-      this.muteSubscriberStream(subId, sub.muteVideo, sub.muteAudio);
+  disableDefaultHandlers() {
+    const disabledHandlers = global.config.erizo.disabledHandlers;
+    if (!this.mediaStream) {
+      return;
+    }
+    for (const index in disabledHandlers) {
+      this.mediaStream.disableHandler(disabledHandlers[index]);
+    }
+  }
+
+  _onSchemeSlideShowModeChange(message, clientId) {
+    this.setSlideShow(message.enabled, clientId);
+  }
+
+  onSignalingMessage(msg) {
```

Comment:
👍 to having the publisher and the subscriber negotiation separated. We can probably have the common parts in `NodeClass` at some point but just not having them inside an `if` makes it a lot clearer